src/hotspot/share/opto/castnode.cpp

```diff
@@ -279,9 +274,9 @@ void CastIINode::dump_spec(outputStream* st) const {
 #endif
 
 CastIINode* CastIINode::pin_array_access_node() const {
-  assert(_dependency == RegularDependency, "already pinned");
+  assert(depends_only_on_test(), "already pinned");
```

Comment:
Would this not be more readable?

```suggestion
  assert(is_dependency_floating(), "already pinned");
```

Reply:
Because it seems we are talking about floating vs pinned here. Adding yet another concept of "depending only on test" would require further explanation / definition.